sotodlib/preprocess/core.py

```diff
@@ -101,19 +109,39 @@ def select(self, meta):
 
         Arguments
         ---------
-        meta : AxisManager 
+        meta : AxisManager
             Metadata related to the specific observation
 
         Returns
         -------
-        meta : AxisManager 
+        meta : AxisManager
             Metadata where non-selected detectors have been removed
         """
 
         if self.select_cfgs is None:
             return meta
         raise NotImplementedError
-    
+
+    def record_qa(self, meta, proc_aman):
+        """ Record a QA metric from the output of this process.
+
+        Arguments
+        ---------
+        meta : AxisManager
+            Metadata related to the specific observation
+        proc_aman : AxisManager
+            The output of the preprocessing pipeline.
+        """
+
+        if self.qa_cfgs is None or self.monitor is None:
+            return
+        metric = self._gen_qa(meta, proc_aman)
+        self.monitor.record(**metric, measurement=self._influx_meas)
```

Comment:
Are you sure these lines won't change in the future / won't constrain posting different measurements? I think you're basically just pushing the definition of all the kwargs to the `_gen_qa` function so it's fine, but just double checking.